specification/compute/resource-manager/Microsoft.Compute/stable/2020-12-01/disk.json

```diff
@@ -0,0 +1,3432 @@
+          },
+          "description": "Resource tags"
+        },
+        "sku": {
+          "$ref": "#/definitions/SnapshotSku"
+        }
+      },
+      "description": "Snapshot update resource."
+    },
+    "SnapshotList": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Snapshot"
+          },
+          "description": "A list of snapshots."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "The uri to fetch the next page of snapshots. Call ListNext() with this to fetch the next page of snapshots."
+        }
+      },
+      "required": [
+        "value"
+      ],
+      "description": "The List Snapshots operation response."
+    },
+    "EncryptionSetIdentity": {
+      "properties": {
+        "type": {
+          "type": "string",
+          "enum": [
+            "SystemAssigned",
+            "None"
+          ],
+          "x-ms-enum": {
+            "name": "DiskEncryptionSetIdentityType",
+            "modelAsString": true
+          },
+          "description": "The type of Managed Identity used by the DiskEncryptionSet. Only SystemAssigned is supported for new creations. Disk Encryption Sets can be updated with Identity type None during migration of subscription to a new Azure Active Directory tenant; it will cause the encrypted resources to lose access to the keys."
+        },
+        "principalId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The object id of the Managed Identity Resource. This will be sent to the RP from ARM via the x-ms-identity-principal-id header in the PUT request if the resource has a systemAssigned(implicit) identity"
+        },
+        "tenantId": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The tenant id of the Managed Identity Resource. This will be sent to the RP from ARM via the x-ms-client-tenant-id header in the PUT request if the resource has a systemAssigned(implicit) identity"
+        }
+      },
+      "description": "The managed identity for the disk encryption set. It should be given permission on the key vault before it can be used to encrypt disks."
+    },
+    "DiskEncryptionSet": {
+      "properties": {
+        "identity": {
+          "$ref": "#/definitions/EncryptionSetIdentity"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/EncryptionSetProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ],
+      "description": "disk encryption set resource."
+    },
+    "DiskEncryptionSetUpdate": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DiskEncryptionSetUpdateProperties"
+        },
+        "identity": {
+          "$ref": "#/definitions/EncryptionSetIdentity"
```

Comment:
when adding a new optional property, please add it to the end of the property list.